pkg/sql/plan2/function/builtin/unary/lengthutf8.go

```diff
@@ -23,21 +24,38 @@ import (
 	"github.com/matrixorigin/matrixone/pkg/vm/process"
 )
 
+var (
+	errorParameterIsNotString = errors.New("the parameter is not char or varchar")
+	errorParameterIsInvalid   = errors.New("invalid parameter")
+)
+
 func LengthUTF8(vectors []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	if len(vectors) == 0 || proc == nil {
+		return nil, errorParameterIsInvalid
+	}
+	if vectors[0] == nil {
+		return nil, errorParameterIsInvalid
+	}
 	inputVector := vectors[0]
 	resultType := types.Type{Oid: types.T_uint64, Size: 8}
 	resultElementSize := int(resultType.Size)
 	if inputVector.IsScalar() {
 		if inputVector.ConstVectorIsNull() {
 			return proc.AllocScalarNullVector(resultType), nil
 		}
-		inputValues := inputVector.Col.(*types.Bytes)
+		inputValues, ok := inputVector.Col.(*types.Bytes)
+		if !ok {
```

Comment:
this check is redundant because the TypeCheckFn will guarantee the input type is of char/varchar

Reply:
In unit test case,  TypeCheckFn will not be called to ensure the correctness of the parameters.
In such assumption, it is better to consider the mismach of the type.

Reply:
okay, then you may need to add type assertion check in quite a lot of code... anything I can help with?
 I think I left a few lines of futile code when fixing conflicts in unary/length.go, feel free to delete it

Reply:
No, I will not do the check in the code out of my plan. That is others' business.

`length` is on xiliang's  plan. he will take care of it.

Reply:
> okay, then you may need to add type assertion check in quite a lot of code... anything I can help with? I think I left a few lines of futile code when fixing conflicts in unary/length.go, feel free to delete it

No, I will not do the check in the code out of my plan. That is others' business.

`length` is on xiliang's  plan. he will take care of it.